automation/orchestrator.go

```diff
@@ -191,41 +209,91 @@ func (o *Orchestrator) Shutdown() {
 	o.cancel()
 }
 
+// advanceTrigger may admit log errors
+func (o *Orchestrator) advanceTrigger(wf *workflow.Workflow, triggerID string) *workflow.Workflow {
+	for i, triggerOpt := range wf.Triggers {
+		trigType := triggerOpt["type"].(string)
+		listener, ok := o.listeners[trigType]
+		if !ok {
+			log.Debugw("advanceTrigger: listener not found", "type", trigType)
+			return wf
+		}
+		trig, err := listener.ConstructTrigger(triggerOpt)
+		if err != nil {
+			log.Debugw("advanceTrigger: error constructing trigger", "error", err)
+			return wf
+		}
+		if trig.ID() == triggerID {
+			trig.Advance()
+			w := wf.Copy()
+			w.Triggers[i] = trig.ToMap()
+			return w
+		}
+	}
+	return wf
+}
+
 // handleTrigger calls `RunWorkflow` when an `event.ETWorkflowTrigger` event is fired
 // it expects the payload for the `event.ETWorkflowTrigger` to be a workflow.ID
 // represented as a string
 func (o *Orchestrator) handleTrigger(ctx context.Context, e event.Event) error {
 	if e.Type == event.ETWorkflowTrigger {
-		wid, ok := e.Payload.(string)
+		wtp, ok := e.Payload.(*event.WorkflowTriggerEvent)
```

Comment:
if we switch payloads from pointers to literals, be sure to update assertions like these. It's a good measure of the test suite: such a change should blow up tests